app/client/keybase/keyvault.go

```diff
@@ -0,0 +1,296 @@
+	if err != nil {
+		return err
+	}
+
+	data := map[string]interface{}{
+		"key_pair": string(keyPairBytes),
+		"hint":     hint,
+	}
+
+	_, err = vk.client.KVv2(vk.mount).Put(context.TODO(), keyPair.GetAddressString(), data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Import a new keypair from the private key hex string provided into Vault
+func (vk *VaultKeybase) ImportFromString(privStr, passphrase, hint string) error {
+	keyPair, err := crypto.CreateNewKeyFromString(privStr, passphrase, hint)
+	if err != nil {
+		return err
+	}
+
+	keyPairBytes, err := json.Marshal(keyPair)
+	if err != nil {
+		return err
+	}
+
+	data := map[string]interface{}{
+		"key_pair": string(keyPairBytes),
+		"hint":     hint,
+	}
+
+	_, err = vk.client.KVv2(vk.mount).Put(context.TODO(), keyPair.GetAddressString(), data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Import a new keypair from the JSON string of the encrypted private key into Vault
+func (vk *VaultKeybase) ImportFromJSON(jsonStr, passphrase string) error {
+	keyPair, err := crypto.ImportKeyFromJSON(jsonStr, passphrase)
+	if err != nil {
+		return err
+	}
+
+	keyPairBytes, err := json.Marshal(keyPair)
+	if err != nil {
+		return err
+	}
+
+	data := map[string]interface{}{
+		"key_pair": string(keyPairBytes),
+	}
+
+	_, err = vk.client.KVv2(vk.mount).Put(context.TODO(), keyPair.GetAddressString(), data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Get a keypair from Vault
+func (vk *VaultKeybase) Get(address string) (crypto.KeyPair, error) {
+	data, err := vk.client.KVv2(vk.mount).Get(context.TODO(), address)
+	if err != nil {
+		return nil, err
+	}
+
+	if data == nil {
+		return nil, errors.New("key not found")
+	}
+
+	keyPairStr, ok := data.Data["key_pair"].(string)
+	if !ok {
+		return nil, errors.New("invalid key pair")
+	}
+
+	var keyPairStruct crypto.EncKeyPair
+	err = json.Unmarshal([]byte(keyPairStr), &keyPairStruct)
+	if err != nil {
+		return nil, err
+	}
+
+	return &keyPairStruct, nil
+}
+
+// Get a public key from Vault
+func (vk *VaultKeybase) GetPubKey(address string) (crypto.PublicKey, error) {
+	keyPair, err := vk.Get(address)
+	if err != nil {
+		return nil, err
+	}
+
+	return keyPair.GetPublicKey(), nil
+}
+
+// Get a private key from Vault
+func (vk *VaultKeybase) GetPrivKey(address, passphrase string) (crypto.PrivateKey, error) {
+	keyPair, err := vk.Get(address)
+	if err != nil {
+		return nil, err
+	}
+
+	privKey, err := keyPair.Unarmour(passphrase)
+	if err != nil {
+		return nil, err
+	}
+
+	return privKey, nil
+}
+
+// Get all keypairs from Vault
+func (vk *VaultKeybase) GetAll() ([]string, []crypto.KeyPair, error) {
+	data, err := vk.client.Logical().List(fmt.Sprintf("%s/metadata", vk.mount))
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if data == nil {
+		return nil, nil, nil
+	}
+
+	var addresses []string
+	var keyPairs []crypto.KeyPair
+
+	for _, key := range data.Data["keys"].([]interface{}) {
```

Comment:
1. s/interface{}/any as of go 1.18
2. Why do we even need a typecast here? I feel like skipping it would avoid the need for string type assertions below